### ISSUE TITLE
Add Profile functionality

### ISSUE THREAD

**Issue by georgiana-b** [frictionlessdata/datapackage-rb#31]
# Overview

Both the [the specs](http://specs.frictionlessdata.io/profiles/) and the [proposed interface](https://github.com/frictionlessdata/implementations#datapackage) refer to the ability to assign a cetrain profile to a datapackage.
Find out  the use-case for it and if it should be implemented.

# Tasks

- [ ] https://github.com/frictionlessdata/datapackage-rb/issues/31#issuecomment-310603664
- [ ] update `DataPackage` and `validate` to re-use `Profile` class

**Comment by roll** [comment]
@georgiana-b 
There is a reference implementation:
- https://github.com/frictionlessdata/datapackage-js/blob/master/src/datapackage.js
- https://github.com/frictionlessdata/datapackage-js/blob/master/src/profile.js

So conceptually it's really simple data package or resource descriptor could have `profile` property:
- with id like `tabular-data-package` from https://specs.frictionlessdata.io/schemas/registry.json
- with a direct path/url to jsonschema

You have a `Profile` class which accept this input and provide `Profile.validate(descriptor)` method. So you data package/resource could validate its descriptor against given profile.

Also profile is exposed via `profile` property:
```py
dp = DataPackage({profile: 'tabular-data-package', ...})
dp.profile.name # tabular-data-package
dp.profile.jsonschema # dict from https://specs.frictionlessdata.io/schemas/tabular-data-package.json
dp.profile.validate(descriptor) # user could validate some other descriptor against a profile of this dp
```

**Comment by roll** [comment]
@georgiana-b 
Alsmost everything you need to finish this one is in https://github.com/frictionlessdata/datapackage-js/pull/76

**Comment by georgiana-b** [comment]
Closed via #43.
